project/profiles/capacity4more/modules/c4m/og/c4m_og_invite_visitors/c4m_og_invite_visitors.module

```diff
@@ -0,0 +1,395 @@
+<?php
+
+/**
+ * @file
+ * Invite new members to a group or project.
+ */
+
+/**
+ * Implements hook_menu().
+ */
+function c4m_og_invite_visitors_menu() {
+  $items = array();
+  $items['group/%/%/admin/members/invite'] = array(
+    'page callback' => 'drupal_get_form',
+    'title' => 'Invite members',
+    'page arguments' => array('_c4m_og_invite_visitors_new_users_form', 1, 2),
+    'type' => MENU_LOCAL_TASK,
+    'weight' => 4,
+    'access callback' => 'c4m_og_invite_visitors_user_access',
+    'access arguments' => array('invite members', 1, 2),
+  );
+
+  return $items;
+}
+
+/**
+ * Implements hook_og_ui_get_group_admin().
+ */
+function c4m_og_invite_visitors_og_ui_get_group_admin($group_type, $gid) {
+  $items = array();
+  if (og_user_access($group_type, $gid, 'invite members')) {
+    $items['c4m_og_invite_visitors'] = array(
+      'title' => t('Invite members'),
+      'description' => t('Create new users and add them as group members.'),
+      'href' => 'admin/members/invite',
+    );
+  }
+
+  return $items;
+}
+
+/**
+ * Check if entity is a group, and user has permission - Access.
+ */
+function c4m_og_invite_visitors_user_access($perm, $group_type, $gid) {
+  global $user;
+
+  $group = entity_load_single($group_type, $gid);
+  if (!$group || !og_is_group($group_type, $group)) {
+    return FALSE;
+  }
+
+  // SAs, GOs and GAs can invite new members.
+  if (_c4m_features_og_members_is_power_user($group)) {
+    return TRUE;
+  }
+
+  // Get request type (open/moderated).
+  $membership_request_value = c4m_og_get_group_membership_request($group);
+  // Get access type (public/private).
+  $access_type = c4m_og_get_access_type($group);
+
+  // For private or moderated groups, return false. Else check for member.
+  if ($access_type['type'] == 'private' || $membership_request_value == 'moderated') {
+    return FALSE;
+  }
+  else {
```

Comment:
we don't need the else in here, please drop it and just return the `og_is_membership()`.